File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int area = 0;
        int l=0;
        int r=height.length-1;
        while(l<r){
            int w = r-l;
            int h = Math.min(height[l], height[r]);
            int temp = w*h;
            area = Math.max(area, temp);
            if(height[l]<height[r]) l++; else r--; 
        }
        return area;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r, area = 0, len(height)-1, 0
        while l<r:
            w = r-l
            h = min(height[r], height[l])
            temp = w*h
            area = max(area, temp)
            if height[l] < height[r]: l += 1
            else: r -= 1
        return area